File: C1.UWP.FlexChart/CS/FlexChart101/DataCreator.cs
using System;
using System.Collections.Generic;
using C1.Chart;

namespace FlexChart101
{
    class DataCreator
    {
        public delegate double MathActionDouble(double num);
        public delegate double MathActionInt(int num);

        public static Dictionary<ChartType, string> CreateSimpleChartTypes()
        {
            Dictionary<ChartType, string> types = new Dictionary<ChartType, string>();
            types = new Dictionary<ChartType, string>();
            types.Add(ChartType.Column, Strings.Column);
            types.Add(ChartType.Bar, Strings.Bar);
            types.Add(ChartType.Line, Strings.Line);
            types.Add(ChartType.Scatter, Strings.Scatter);
            types.Add(ChartType.LineSymbols, Strings.LineSymbols);
            types.Add(ChartType.Area, Strings.Area);
            types.Add(ChartType.Spline, Strings.Spline);
            types.Add(ChartType.SplineArea, Strings.SplineArea);
            types.Add(ChartType.SplineSymbols, Strings.SplineSymbols);
            types.Add(ChartType.Step, Strings.Step);
            types.Add(ChartType.StepArea, Strings.StepArea);
            types.Add(ChartType.StepSymbols, Strings.StepSymbols);

            return types;
        }

        public static List<FruitDataItem> CreateFruit()
        {
            var fruits = new string[] { "Oranges", "Apples", "Pears", "Bananas" };
            var count = fruits.Length;
            var result = new List<FruitDataItem>();
            var rnd = new Random();
            for (var i = 0; i < count; i++)
                result.Add(new FruitDataItem()
                {
                    Fruit = fruits[i],
                    March = rnd.Next(20),
                    April = rnd.Next(20),
                    May = rnd.Next(20),
                });
            return result;
        }

        public static List<DataItem> CreateData()
        {
            var data = new List<DataItem>();
            data.Add(new DataItem("UK", 5, 4));
            data.Add(new DataItem("USA", 7, 3));
            data.Add(new DataItem("Germany", 8, 5));
            data.Add(new DataItem("Japan", 12, 8));
            return data;
        }

        public static List<GroupDataItem> CreateGroupData()
        {
            var data = new List<GroupDataItem>();
            data.Add(new GroupDataItem("UK", 5, 4, 6, 5));
            data.Add(new GroupDataItem("USA", 7, 6, 3, 2));
            data.Add(new GroupDataItem("Germany", 8, 5, 10, 6));
            data.Add(new GroupDataItem("Japan", 12, 8, 10, 7));
            return data;
        }

        public static List<DataItem> CreateFunnelData()
        {
            var data = new List<DataItem>();
            var countries = "US,Germany,UK,Japan,Italy,Greece".Split(',');
            var sales = 10000;
            var rnd = new Random();
            for (var i = 0; i < countries.Length; i++)
            {
                var item = new DataItem(countries[i], sales, 0);
                sales = sales - (int)Math.Round(rnd.NextDouble() * 2000);
                data.Add(item);
            }

            return data;
        }
    }

    public class FruitDataItem
    {
        public string Fruit { get; set; }
        public double March { get; set; }
        public double April { get; set; }
        public double May { get; set; }
    }

    public class DataItem
    {
        public DataItem(string country, int sales, int expenses)
        {
            Country = country;
            Sales = sales;
            Expenses = expenses;
        }

        public string Country { get; set; }
        public int Sales { get; set; }
        public int Expenses { get; set; }
    }

    public class GroupDataItem
    {
        public GroupDataItem(string country, int domesticSales, int domesticExpenses, int exportSales, int exportExpenses)
        {
            Country = country;
            DomesticSales = domesticSales;
            DomesticExpenses = domesticExpenses;
            ExportSales = exportSales;
            ExportExpenses = exportExpenses;
        }

        public string Country { get; set; }
        public int DomesticSales { get; set; }
        public int DomesticExpenses { get; set; }
        public int ExportSales { get; set; }
        public int ExportExpenses { get; set; }
    }

    public class DynamicItem
    {
        public int Day { get; set; }
        public int Trucks { get; set; }
        public int Ships { get; set; }
        public int Planes { get; set; }
    }
}

File: C1.UWP.FlexChart/VB/FlexChart101/DataCreator.vb
Imports C1.Chart
Imports System.Collections.Generic
Imports System

Class DataCreator
    Public Delegate Function MathActionDouble(num As Double) As Double
    Public Delegate Function MathActionInt(num As Integer) As Double

    Public Shared Function CreateSimpleChartTypes() As Dictionary(Of ChartType, String)
        Dim types As New Dictionary(Of ChartType, String)()
        types = New Dictionary(Of ChartType, String)()
        types.Add(ChartType.Column, Strings.Column)
        types.Add(ChartType.Bar, Strings.Bar)
        types.Add(ChartType.Line, Strings.Line)
        types.Add(ChartType.Scatter, Strings.Scatter)
        types.Add(ChartType.LineSymbols, Strings.LineSymbols)
        types.Add(ChartType.Area, Strings.Area)
        types.Add(ChartType.Spline, Strings.Spline)
        types.Add(ChartType.SplineArea, Strings.SplineArea)
        types.Add(ChartType.SplineSymbols, Strings.SplineSymbols)
        types.Add(ChartType.Step, Strings.StepType)
        types.Add(ChartType.StepArea, Strings.StepArea)
        types.Add(ChartType.StepSymbols, Strings.StepSymbols)

        Return types
    End Function

    Public Shared Function CreateFruit() As List(Of FruitDataItem)
        Dim fruits As String() = {"Oranges", "Apples", "Pears", "Bananas"}
        Dim count As Integer = fruits.Length
        Dim result As New List(Of FruitDataItem)()
        Dim rnd As New Random()
        Dim i As Integer = 0
        While i < count
            result.Add(New FruitDataItem() With {
                .Fruit = fruits(i),
                .March = rnd.[Next](20),
                .April = rnd.[Next](20),
                .May = rnd.[Next](20)
            })
            i += 1
        End While
        Return result
    End Function

    Public Shared Function CreateData() As List(Of DataItem)
        Dim data As New List(Of DataItem)()
        data.Add(New DataItem("UK", 5, 4))
        data.Add(New DataItem("USA", 7, 3))
        data.Add(New DataItem("Germany", 8, 5))
        data.Add(New DataItem("Japan", 12, 8))
        Return data
    End Function

    Public Shared Function CreateGroupData() As List(Of GroupDataItem)
        Dim data As New List(Of GroupDataItem)()
        data.Add(New GroupDataItem("UK", 5, 4, 6, 5))
        data.Add(New GroupDataItem("USA", 7, 6, 3, 2))
        data.Add(New GroupDataItem("Germany", 8, 5, 10, 6))
        data.Add(New GroupDataItem("Japan", 12, 8, 10, 7))
        Return data
    End Function

    Public Shared Function CreateFunnelData() As List(Of DataItem)
        Dim data As New List(Of DataItem)
        Dim countries As String() = "US,Germany,UK,Japan,Italy,Greece".Split(",")
        Dim sales As Integer = 10000
        Dim rnd As Random = New Random()
        Dim i As Integer = 0
        While i < countries.Length
            Dim item As New DataItem(countries(i), sales, 0)
            sales = sales - CType(Math.Round(rnd.NextDouble() * 2000), Integer)
            data.Add(item)
            i += 1
        End While
        Return data
    End Function
End Class

Public Class FruitDataItem
    Public Property Fruit() As String
    Public Property March() As Double
    Public Property April() As Double
    Public Property May() As Double
End Class

Public Class DataItem
    Public Sub New(c As String, s As Integer, e As Integer)
        Country = c
        Sales = s
        Expenses = e
    End Sub

    Public Property Country() As String
    Public Property Sales() As Integer
    Public Property Expenses() As Integer
End Class

Public Class GroupDataItem
    Public Sub New(c As String, ds As Integer, de As Integer, es As Integer, ee As Integer)
        Country = c
        DomesticSales = ds
        DomesticExpenses = de
        ExportSales = es
        ExportExpenses = ee
    End Sub

    Public Property Country() As String
    Public Property DomesticSales() As Integer
    Public Property DomesticExpenses() As Integer
    Public Property ExportSales() As Integer
    Public Property ExportExpenses() As Integer
End Class


Public Class DynamicItem
    Public Property Day() As Integer
    Public Property Trucks() As Integer
    Public Property Ships() As Integer
    Public Property Planes() As Integer
End Class